### ISSUE TITLE
Missing Devices in local API but ok in cloud API.

### ISSUE THREAD

**Issue by qelanhari** [Somfy-Developer/Somfy-TaHoma-Developer-Mode#20]
Hi !
I'm trying to use the Local API. 
No issue using the endpoints but some of my IO devices are not returned from the local API when they are from the cloud API. 
These are officially supported device (Atlantic ZoneControl 2.0) that I can control from the Tahoma App.
I can also control them via Cloud API. They are simply not returned by the local API using the `/setup/devices` endpoint.
I only get my Sliding Gate and my Garage Door and surprisingly my unsupported WaterHeater (Atlantic Calypso).

Thank you for the feedback.

**Comment by adriweb** [comment]
Not all protocols and devices are supported by the local mode.

**Comment by qelanhari** [comment]
Is this a temporary restriction ?
Can we have details on the restriction critiria to know what to expect ?

**Comment by Minasu** [comment]
Hello,

No, the heating equipment is not seen by the local API and that is very unfortunate.

https://github.com/Somfy-Developer/Somfy-TaHoma-Developer-Mode/issues/11

**Comment by nyroDev** [comment]
Hello @Somfy-Developer
Is there an ETA or similar to have all devices accessible locally please?

**Comment by iMicknl** [comment]
+1! It would be great if all devices could be exposed, otherwise there is still need for a hybrid approach where local and cloud API are used together..

**Comment by HpNoTiQ56** [comment]
+1 : I'd loved that my atlantic electric heater could use local API.
ATM, I'm using both local and cloud integration in Home Assistant.
Any ETA or won't fix status?

**Comment by nyroDev** [comment]
After update 2022.6.4-6, still no new devices accessible locally (at least for me)

**Comment by Quentame** [comment]
@qelanhari : I think you should edit the title to precise it concerns the local API.
"Missing devices on local API" ?

**Comment by qelanhari** [comment]
This whole repo only concerns local api.

**Comment by HpNoTiQ56** [comment]
Any news on this one?
Can't use local and cloud integration at the same time!

**Comment by Tronix117** [comment]
I'm also adding my voice to this. I can't understand why it's restricted this way. What is the meaning of having the need to go through the cloud when everything can be handled locally...
For now I' really starting to be regretful of choosing IO for some of my devices. So far only restrictions over restrictions due to this troublesome closed protocol (even if you say it's opened), it does not represent the future of home automation.
I hope matter compatibility will be handled in a better way

**Comment by MacDoofy** [comment]
@llavorel-somfy seems you are the official representative here for the Tahoma developer mode. Would you have any information on adding the climate control feature to this ?

**Comment by daahmer** [comment]
I'm also disapointed that heaters are not available on local API.
each time that Overkiz servers are unavailable or when I lose my internet connexion, I cannot pilot my heaters.
I recently changed my bridge (from cozytouch to tahoma switch) because there was local API on tahoma switch. at no time did I suspect that the APIs weren't complete.
Could you please give us some visibility on the fix for this issue please.

**Comment by Tronix117** [comment]
@llavorel-somfy hello, could you help us understand why the reason behind that ?

Is it :
1. it's on the roadmap, not just there yet !
2. no time/budget available to support this
3. technicaly complicated, it's not just a filter to remove
4. legal reasons (no need to provide further details if they can't be disclosed)

**Comment by Gandulf78** [comment]
I am also very interested in this local API feature for heaters. Any progress in Somfy roadmap? 
In 2024, we shouldn't rely only on the cloud for heaters.

**Comment by SherekhanFR** [comment]
Same here, this is very disappointing to not have all the devices, especially the heaters / heat pumps / boilers, manageable by the local API. I wanted to dig on this today to build a script that would check the EDF Tempo color and turn on/off accordingly my heat pump. I feel helpless to achieve this unless I change for another home management system. If beyond someone at Somfy to give us light on the Tahoma API roadmap, if anyone has alternative for my topic, please do not hesitate.

**Comment by jojolll** [comment]
Same here. The lack of full control over essential devices like boilers through the local API is frustrating. It really doesn’t seem like it would be complicated to implement, but they just don’t seem willing to make the effort.

**Comment by qelanhari** [comment]
Hi @llavorel-somfy !

Original author here ! As I see you are still actively working on the developer API, I have to ask again regarding this subject. Here I just need devices that are already in the cloud API.

I could imagine that adding new devices on the API is a pain, but for ZoneControl I'd guess this is just a matter of adding those types/class into a declarative list or copy/pasting some lines of existing code ?

(Pretty) Please at least give us a feedback on the matter, it's been more that 2 years since this issue has been opened without any interaction beside the "enhancement" label added (which is better than a "no fix" for sure).

Thanks and have a nice weekend.

**Comment by llavorel-somfy** [comment]
Hi @qelanhari 

The topic is well selected as a possible evolution of our API. The development team backlog is dense and this evolution has not been planned yet.

**Comment by Tronix117** [comment]
Thank you for your feedback, that is at least nice to hear it's not abandoned !

Lot of people are expecting local API to work with their devices (in other communities), it improves a lot the reliability of the whole installation as well as the execution delays.

Can't wait to see some progress on that :) !

**Comment by llavorel-somfy** [comment]
You're welcome !

We try to keep the different issues up to date so if you don't see an explicit "won't fix", it means the enhancement is still in our radar.

**Comment by hugochar** [comment]
If someone is interested or trying to get is water heater in local api, it works, i've got my aquacosy from atlantic in home assistant in local mode connected via api

<img width="541" alt="Image" src="https://github.com/user-attachments/assets/9aef8acd-2ccb-427b-89a3-b83ee495acd3" />

**Comment by hugochar** [comment]
@llavorel-somfy  Do you have any news regarding the solving of the problem ?

**Comment by Loic691** [comment]
+1

**Comment by llavorel-somfy** [comment]
> [@llavorel-somfy](https://github.com/llavorel-somfy) Do you have any news regarding the solving of the problem ?

@hugochar  could you be explicit on the devices you are missing please ?

**Comment by hugochar** [comment]
@llavorel-somfy 

Same as the author, my heaters, i'v got everything except them

they're in tahoma 

![Screenshot_20250613-195124.png](https://github.com/user-attachments/assets/00c47862-b003-496b-8c8d-f3787b7e1fa9)


but not in home assistant in local api

![Screenshot_20250613-195157.png](https://github.com/user-attachments/assets/16cbbac8-70b0-42af-9657-c670a682ed3f)

**Comment by hugochar** [comment]
Any news ?

**Comment by llavorel-somfy** [comment]
Hi @hugochar 

This feature is still in our product backlog. We have planned to work on it soon. I will change the label when it is ready.

**Comment by hugochar** [comment]
Great news 💪👌

**Comment by laurentftech** [comment]
Great news indeed. 
Is it plan to cover the following HW  by any chance?
Thank you. 

ovp:SomfyHeatingTemperatureInterfaceOVPComponent (central thermostat)

io:AtlanticElectricalHeaterWithAdjustableTemperatureSetpointIOComponent (IPALA Heather)

**Comment by HpNoTiQ56** [comment]
Hi all, it's summer end and winter is coming, may I ask if I will control my heaters using local api for Christmas?

**Comment by hugochar** [comment]
I guess we will not get anything for christmas 😖

**Comment by hugochar** [comment]
@llavorel-somfy do you have some news from Somfy ?

**Comment by HpNoTiQ56** [comment]
@llavorel-somfy Hi Any news on this?

**Comment by qelanhari** [comment]
Hello @llavorel-somfy, it seems that the restrictions on the cloud API have been tightened. I can no longer reliably control my heat pump (Atlantic ZoneControl 2.0). I am experiencing disconnections and lost commands on a daily basis. 
I cannot make automation though the Tahoma app based on presence/occupancy, neither using the physical remotes. 
It would really help to have them in the local API, a really nice Christmas Gift I'd say.

Thanks again.

**Comment by hugochar** [comment]
I am watching for an alternative as somfy isn't doing the job, it seams that the rexel box energeasy Connect does the job pretty well and have a local api ( confirmed by a tech from rexel by phone).

i'm gonna give it a try

**Comment by qelanhari** [comment]
It's a rebranded Tahoma Switch/DIN, I'd guess its Somfy's firmware inside, therefor same limitations :)

**Comment by hugochar** [comment]
yep, i'll see

**Comment by rhessus** [comment]
I’ve just made this unfortunate discovery… I have a Cozytouch bridge, with a Thermor water heater and 3 Sauter radiators. After pairing everything on the TaHoma Switch, I’ve just realized that none of these devices show up via the API… I’m really pissed off (so it’s basically useless). On top of that, when I tried to put everything back on CozyTouch, it completely broke my setup for my 6 roller shutters… Great day.

**Comment by hugochar** [comment]
Welcome and take a seat, we are all facing that and waiting for a miracle 🙏

**Comment by bagasse** [comment]
If you expect something from somfy or its ecosystem, you may wait for a long time. This ecosystem is broken, and somfy don't care about their users, they run on a captive market. I've mostly the same issue @rhessus, I have to run a cozytouch (which is the worst box / application that I've ever used), tahoma and Tydom (the more open and usable by far for the three, even if it's also a closed ecosystem) boxes in parallel and Home assistant on top of it. I hoped to at least merge the two IO boxes and run only the tahoma, they both talk IO home control after all, but it's not possible because my atlantic optimocosy HR plus are not recognized. They come from the old world, they don't understand that openness and  interoperability is the key.

So yes, apart from a miracle, I couldn't see was could happen...

**Comment by gonzague** [comment]
Hey @llavorel-somfy @JanJaapKo @FranzDiebold @tetienne @vhenriet-sfy .. People are complaining all over the Internet about Somfy abandonning the developper API.

I was advocating the Tahoma box recently and man.. the backlash. 

Is something going to be done about it ? This casts a bad image on what was a good .. idea. People are trying to move away from what felt like a good thing coming from Somfy.

@hugochar Apparently the Rexel is the OEM for Somfy :)  / @qelanhari it seems to be the opposite actually . But it is worth going to check their Zendesk https://assistance.energeasyconnect.com/hc/fr/community/topics/360000705140-Applications-Energeasyconnect

**Comment by rhessus** [comment]
I’ve just received an email saying that version 2.3 is coming on January 13, 2026… maybe the miracle will happen!! :D Since we can’t know what’s hidden behind this update, suspense!

**Comment by hugochar** [comment]
Guys ?

I've my water heater that show up localy in tahoma and home assistant 

![image](https://github.com/user-attachments/assets/e3f418f7-da81-4698-adb7-7ef52a06d5c4)

**Comment by qelanhari** [comment]
Its been few weeks(months) already. Mine is appearing twice 🙃

Edit : in local api, as mentioned in the original issue, it has always been the case. But in the tahoma app, its recent.

**Comment by hugochar** [comment]
Mine show up twice also did'nt see it before btw

**Comment by hugochar** [comment]
> Its been few weeks(months) already. Mine is appearing twice 🙃
> 
> Edit : in local api, as mentioned in the original issue, it has always been the case. But in the tahoma app, its recent.

It's could be a sign that they're working on it
